NanoAnalysis/python/modules/muonScaleRes.py

```diff
@@ -0,0 +1,85 @@
+from __future__ import print_function
+from PhysicsTools.NanoAODTools.postprocessing.framework.eventloop import Module
+from PhysicsTools.NanoAODTools.postprocessing.framework.datamodel import Collection
+import os
+import numpy as np
+from math import pi
+from ROOT import MuonScaRe
+
+class muonScaleRes(Module):
+    def __init__(self, json, is_mc, overwritePt=False):
+        """Add branches for muon scale and resolution corrections.
+        Parameters:
+            json: full path of json file
+            is_mc: True for MC (smear pt+add uncertainties), False for data (scale pt)
+            overwritePt: replace value in the pt branch, and store the old one as "uncorrected_pt"
+        """
+
+        self.overwritePt = overwritePt
+        self.is_mc = is_mc
+        
+        self.corrModule = MuonScaRe(json)
+
+
+    def getPtCorr(self, event, muon, var = "nom") :
```

Comment:
`event` as argument of this function can be removed, since it's no longer used.

Probably we can add here (before L26) an `if muon.pt > 200: return muon.pt` so that the corrections are not applied to high pT muons, as per POG recommendation.

Reply:
Done. Indeed it solves the problem, the correction seems OK now.